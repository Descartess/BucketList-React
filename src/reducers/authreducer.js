const INITIAL_STATE = {
  loading: false,
  token: null,
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case 'REGISTER_PENDING':
      return { ...state, loading: true };
    case 'REGISTER_FULFILLED':
      return { ...state, loading: false, token: action.payload.data.token };
    case 'SIGNIN_PENDING':
      return { ...state, loading: true };
    case 'SIGNIN_FULFILLED':
      return { ...state, loading: false, token: action.payload.data.token };
    default:
      return state;
  }
};
