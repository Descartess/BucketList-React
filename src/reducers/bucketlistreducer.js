const INITIAL_STATE = {
  bucketlists: [],
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case 'GET_BUCKETLISTS_PENDING':
      return { ...state };
    case 'GET_BUCKETLISTS_FULFILLED':
      return { ...state, bucketlists: action.payload.data.bucketlists };
    case 'POST_BUCKETLISTS_PENDING':
      return { ...state };
    case 'POST_BUCKETLISTS_FULFILLED':
      return { ...state };
    default:
      return { ...state };
  }
};
