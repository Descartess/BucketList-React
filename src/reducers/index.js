import { combineReducers } from 'redux';
import AuthReducer from './authreducer';
import BucketListReducer from './bucketlistreducer';

const reducers = combineReducers({
  auth: AuthReducer,
  bucketlists: BucketListReducer,
});

export default reducers;
