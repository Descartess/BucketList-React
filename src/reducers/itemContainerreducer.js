export const initialState = {
  showEditItem: false,
  showAddItem: false,
  showDeleteItem: false,
  item_name: '',
  completed: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case 'SHOW_EDIT_ITEM':
      return { ...state, showEditItem: !state.showEditItem };
    case 'SHOW_ADD_ITEM':
      return { ...state, showAddItem: !state.showAddItem };
    case 'SHOW_DELETE_ITEM':
      return { ...state, showDeleteItem: !state.showDeleteItem };
    case 'TOGGLE_COMPLETED':
      return { ...state, completed: !state.completed };
    case 'ON_NAME_CHANGE':
      return { ...state, item_name: action.payload };
    default:
      return state;
  }
};
