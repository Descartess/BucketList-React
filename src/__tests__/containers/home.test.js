import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise-middleware';
import renderer from 'react-test-renderer';
import moxios from 'moxios';
import instance from '../../config/axiosconfig';
import Home from '../../containers/home';
import reducers from '../../reducers';

const middleware = applyMiddleware(thunk, promiseMiddleware());

const testStore = createStore(reducers, middleware);

describe('App snapshot ', () => {
  beforeEach(() => {
    moxios.install(instance);
  });
  afterEach(() => {
    moxios.uninstall(instance);
  });
  it('renders correctly', () => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
      });
    });
    const rendered = renderer.create(
      <Provider store={testStore} >
        <Home />
      </Provider>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
