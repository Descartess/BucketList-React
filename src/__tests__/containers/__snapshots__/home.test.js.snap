// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot  renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
        >
          BucketList
        </a>
      </div>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}
      >
        <li
          className="dropdown"
          style={undefined}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href="#"
            id="basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Account Settings
             
            <span
              className="caret"
            />
          </a>
          <ul
            aria-labelledby="basic-nav-dropdown"
            className="dropdown-menu"
            role="menu"
          >
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Sign Out
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Reset Password 
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </nav>
  <div>
    <div
      className="col-md-4"
    >
      <div
        className="panel panel-default"
      >
        <div
          className="panel-heading"
        >
          BucketLists
          <div
            className="pull-right"
            id="addBucket"
            onClick={[Function]}
          >
            <span
              className="glyphicon glyphicon-plus"
            />
             Add
          </div>
        </div>
        <div
          className="panel-body"
        >
          <p>
             No bucketlists created.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="col-md-6"
    >
      <div
        className="panel panel-default"
      >
        <div
          className="panel-heading"
        >
          Items
        </div>
        <div
          className="panel-body"
        >
          <p>
             Click on bucketlist to display items 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
